### PR TITLE
Write one test

### PR DESCRIPTION
**Pull request by andycroll**

### CODE REVIEW

File: _posts/2019/2019-11-03-write-one-test.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+category: ruby
+image:
+  base: '2019/write-one-test'
+  alt: 'Smoke'
+  credit: 'Pascal Meier'
+  source: 'https://unsplash.com/photos/1uVCTVSn-2o'
+
+---
+
+Rails comes with a built-in testing framework and many Rubyists eulogise various methods of testing. Behaviour Driven Development, 100% test coverage, Red-green-refactor.
```
Comment (nodunayo) [inline _posts/2019/2019-11-03-write-one-test.md]: colon after 'testing'

Comment (nodunayo) [reply _posts/2019/2019-11-03-write-one-test.md]: *Red-Green-Refactor

File: _posts/2019/2019-11-03-write-one-test.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+But perhaps you’ve never really ‘got it’ or ever seen the benefits of an established testing culture. As a result you might not have got _any_ tests on your Rails application.
```
Comment (nodunayo) [inline _posts/2019/2019-11-03-write-one-test.md]: *...you might not have _any_ tests _in_ your Rails application.

File: _posts/2019/2019-11-03-write-one-test.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+
+## Instead of...
+
+...not having any tests
```
Comment (nodunayo) [inline _posts/2019/2019-11-03-write-one-test.md]: End with another ellipses, or a comma, or colon..

File: _posts/2019/2019-11-03-write-one-test.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+You can run this test by typing `bundle exec rails test:system`.
+
+
+## Why?
+
+This single test does two major things.
+
+One, because it is a ‘system test’ using a real browser to test your entire site, it checks that your application builds, runs successfully and displays a homepage to the world. This has value.
+
+Two, now you’ve taken your first step into testing, you can take more.
```
Comment (nodunayo) [inline _posts/2019/2019-11-03-write-one-test.md]: 😁 — knew that's where you were going with this!

File: _posts/2019/2019-11-03-write-one-test.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+
+## Why not?
+
+For me, it is a case of “why stop here”. Once you have this basic smoke test, you can add a simple test to check that sign up of new users works. Or that signing in an existing user works.
```
Comment (nodunayo) [inline _posts/2019/2019-11-03-write-one-test.md]: *It's a case of: why stop here?

File: _posts/2019/2019-11-03-write-one-test.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+These sorts of simple, but full system, tests will give you a surprising amount of benefit for very little ongoing effort.
+
+These kinds of system tests rarely allow you to test all the permutations of sophisticated logic within your application, those sorts of things are better tested in unit tests. For general “is my application mostly working”-type concerns you'll get a lot of milage (and coverage) out of these broad smoke tests.
```
Comment (nodunayo) [inline _posts/2019/2019-11-03-write-one-test.md]:
```suggestion
These kinds of system tests rarely allow you to test all the permutations of sophisticated logic within your application. Those sorts of things are better tested in unit tests. For general “is my application mostly working”-type concerns, you'll get a lot of milage (and coverage) out of these broad smoke tests.
```

**Review by nodunayo — DISMISSED** [review]

File: _posts/2019/2019-11-03-write-one-test.md (hunk 1 above)
Comment (andycroll) [reply _posts/2019/2019-11-03-write-one-test.md]:
```suggestion
Rails comes with a built-in testing framework and many Rubyists eulogise various methods of testing: Behaviour Driven Development, 100% test coverage, Red-Green-Refactor.
```

File: _posts/2019/2019-11-03-write-one-test.md (hunk 3 above)
Comment (andycroll) [reply _posts/2019/2019-11-03-write-one-test.md]:
```suggestion
...not having any tests:
```

File: _posts/2019/2019-11-03-write-one-test.md (hunk 2 above)
Comment (andycroll) [reply _posts/2019/2019-11-03-write-one-test.md]:
```suggestion
But perhaps you’ve never really ‘got it’ or ever seen the benefits of an established testing culture. As a result you might not have _any_ tests in your Rails application.
```

File: _posts/2019/2019-11-03-write-one-test.md (hunk 5 above)
Comment (andycroll) [reply _posts/2019/2019-11-03-write-one-test.md]:
```suggestion
For me, it’s a case of: why stop here? Once you have this basic smoke test, you can add a simple test to check that sign up of new users works. Or that signing in an existing user works.
```

**Pull request merged**